### ISSUE TITLE
Menu: Formatting

### ISSUE THREAD

**Issue by easyaspi314** [400plus/400plus#410]
I have to admit: The menus look a LOT better than CHDK. However, could we have these options:
* Scrollbars. These should be showing at all times if the menu scrolls, and, if possible,
* Tabbed menus that look like the XTi's native menu. I can design the icons, just tell me the dimensions.

**Comment by eduperez** [comment]
Those ideas seem interesting. The menu at 400plus is some native menu for the Direct Print functionality, that we reuse for our convenience. We could probably try to move to the "main menus", as you suggested; then we will have the possibility of drawing icons (but I do not see scroll bars there, I am not sure that would be possible). However, this involves a lot of work: we first need to investigate how to handle the "main menu", and then change all of our code to use those menus. Sincerely, I do not think the current menus are that bad to justify all the work needed.

**Comment by easyaspi314** [comment]
Well that explains the native menus. So can `400plus` not do overlays like CHDK can? That stinks. If we could, we could put a manual in the menus, and make it more friendly. I just don't know how it works. My C knowledge comes from Java. So it is awful.

**Comment by 0xAF** [comment]
Edu, IIRC, you can change the menu from "DP Menu" (or whatever we are using now), to "Main Menu" (or whatever you need).
It should be the first parameter of dialog_create (menu.c:56).
I might be wrong on this, but I remember that when we needed some "window" for "help/hints/tips/etc.", I was experimenting with different menu styles. There should be something in the devinfo about the "menu", that I saw usable for help screen.
So basically, if this is correct, you should be able to choose a menu with tabs. Thought it will require something different to be filled with actual submenus, I guess...

**Comment by eduperez** [comment]
@easyaspi314: No, unlike CHDK or ML, menus in 400plus are not drawn using overlays. We tried overlays some time ago (to add some info on the main dialog), but it locked the camera randomly, and we had to deactivate it. I will try new ideas on the overlay for the next release, but I do not think we should convert all the menus to overlays.

@0xAF: Yes, in theory we can open any other menu by just changing the identifier. But the main menu is made of different sub-dialogs, and I think there is a lot of work in moving from our menus to the main menus.

In conclusion, I agree that the current menus still have lots of room for improvement, and using the main menu as a base looks like a cool idea; but I also think that the amount of work needed is not justified. Considering the (slowing down) pace of development, the (negative) influx of new developers and users, the (ever increasing) age of our camera, etc, I think we should be realistic and concentrate all our efforts on the features that can have an impact on the user.

**Comment by 0xAF** [comment]
@eduperez, I agree about the effort it needs. And it seems there is no point in putting this effort in such cosmetic change.

**Comment by easyaspi314** [comment]
OK. Maybe, we could put a ↓, or a "v" if we don't support the first symbol
on the last item, and a ↑ or "^" in the first item when it scrolls.

**Comment by eduperez** [comment]
@easyaspi314: Good idea, that seems easier to implement.

**Comment by nicorikken** [comment]
As a new user took me a couple of minutes to get used to the _inusabilities_. Decent Wiki documentation also provides some of the desired location context. Some improvements would indeed be nice.
